models/members.js

```diff
@@ -115,9 +115,40 @@ const addArchiveRoleToMembers = async (userId) => {
   }
 };
 
+const updateRoles = async (userId, newRoles) => {
+  try {
+    const userDoc = await userModel.doc(userId).get();
+    const user = userDoc.data();
+    let roles;
+    if (Object.keys(newRoles).includes("member")) {
+      if (newRoles.member === true) {
+        if (user?.roles?.member) {
+          return { isRoleUpdated: false };
+        }
+      } else if (newRoles.member === false) {
+        if (user?.roles?.member === false) {
+          return { isRoleUpdated: false };
+        }
+      }
+      roles = user.roles ? { ...user.roles, member: newRoles.member } : { member: newRoles.member };
+    } else if (Object.keys(newRoles).includes("archived")) {
+      if (user?.roles && user.roles[ROLES.ARCHIVED]) return { isRoleUpdated: false };
+      roles = { ...user.roles, archived: newRoles.archived };
```

Comment:
Here also I suggest checking if the role exists similar to the 
```suggestion
      roles = user.roles ? { ...user.roles, archived: newRoles.archived } : { archived: newRoles.archived };
```

Reply:
But everytime when new user created in backend... `archived =false` is already present

Reply:
Done